packages/malloy/src/dialect/functions/malloy_standard_functions.ts

```diff
@@ -0,0 +1,778 @@
+};
+
+const trunc: DefinitionFor<Standard['trunc']> = {
+  'to_integer': {
+    takes: {'value': 'number'},
+    returns: 'number',
+    impl: {function: 'TRUNC'},
+  },
+  // TODO precision should only accept integers, but we don't have a good
+  // way of expressing that constraint at the moment
+  'to_precision': {
+    takes: {'value': 'number', 'precision': 'number'},
+    returns: 'number',
+    impl: {function: 'TRUNC'},
+  },
+};
+
+const unicode: DefinitionFor<Standard['unicode']> = {
+  takes: {'value': 'string'},
+  returns: 'number',
+  impl: {function: 'UNICODE'},
+};
+
+const upper: DefinitionFor<Standard['upper']> = {
+  takes: {'value': 'string'},
+  returns: 'string',
+  impl: {function: 'UPPER'},
+};
+
+// Aggregate functions
+const stddev: DefinitionFor<Standard['stddev']> = {
+  takes: {'value': {dimension: 'number'}},
+  returns: {measure: 'number'},
+  impl: {function: 'STDDEV'},
+};
+
+// Analytic functions
+const avg_moving: DefinitionFor<Standard['avg_moving']> = {
+  'preceding': {
+    generic: ['T', ['string', 'number', 'timestamp', 'date']],
+    takes: {
+      'value': {measure: {generic: 'T'}},
+      'preceding': {literal: 'number'},
+    },
+    returns: {calculation: {generic: 'T'}},
+    impl: {
+      sql: 'AVG(${value})',
+      needsWindowOrderBy: true,
+      between: {preceding: 'preceding', following: 0},
+    },
+  },
+  'following': {
+    generic: ['T', ['string', 'number', 'timestamp', 'date']],
+    takes: {
+      'value': {measure: {generic: 'T'}},
+      'preceding': {literal: 'number'},
+      'following': {literal: 'number'},
+    },
+    returns: {calculation: {generic: 'T'}},
+    impl: {
+      sql: 'AVG(${value})',
+      needsWindowOrderBy: true,
+      between: {preceding: 'preceding', following: 'following'},
+    },
+  },
+};
+
+const first_value: DefinitionFor<Standard['first_value']> = {
+  generic: ['T', ['string', 'number', 'timestamp', 'date', 'json']],
+  takes: {'value': {measure: {generic: 'T'}}},
+  returns: {calculation: {generic: 'T'}},
+  impl: {function: 'FIRST_VALUE', needsWindowOrderBy: true},
+};
+
+const LAG_TYPES: ExpressionValueType[] = [
+  'string',
+  'number',
+  'timestamp',
+  'date',
+  'json',
+  'boolean',
+];
+const lag: DefinitionFor<Standard['lag']> = {
+  'bare': {
+    generic: ['T', LAG_TYPES],
+    takes: {
+      'value': {measure: {generic: 'T'}},
+    },
+    returns: {calculation: {generic: 'T'}},
+    impl: {function: 'LAG', needsWindowOrderBy: true},
+  },
+  'with_offset': {
+    generic: ['T', LAG_TYPES],
+    takes: {
+      'value': {measure: {generic: 'T'}},
+      'offset': {literal: 'number'},
+    },
+    returns: {calculation: {generic: 'T'}},
+    impl: {function: 'LAG', needsWindowOrderBy: true},
+  },
+  'with_default': {
+    generic: ['T', LAG_TYPES],
+    takes: {
+      'value': {measure: {generic: 'T'}},
+      'offset': {literal: 'number'},
+      'default': {constant: {generic: 'T'}}, // TODO needs to be constant max aggregate?
```

Comment:
`LAG` theoretically supports `lag(value, null, count())` (a "constant aggregate" for the default value). This currently doesn't work, because aggregates are currently treated as "output space" expressions, but that conflicts with the "constant" requirement. I haven't worked out exactly what this implies about the Malloy type system. In the IR, this is expressible as `{ type: T, exprType: 'aggregate', evalSpace: 'constant' }`, but but type-checkable in `expr-func`. In the blueprint system, it is not expressible. Since this doesn't change current behavior, I feel okay making this change while I figure out what the right long term answer is, and if necessary, update the blueprint system to allow for it.